editoast/src/views/work_schedules.rs

```diff
@@ -175,15 +181,109 @@ async fn create(
     }))
 }
 
+#[derive(Serialize, Deserialize, ToSchema)]
+struct WorkScheduleProjectForm {
+    work_schedule_group_id: i64,
+    #[schema(value_type = Vec<TrackRange>)]
+    path_track_ranges: Vec<CoreTrackRange>,
+}
+
+/// This struct represents the projection of one work_schedule on a path.
+/// it takes the type, start time, end time of the work_schedule
+/// as well as a list of ranges (a, b),
+/// where a and b are the distance of the beginning and end of the track range
+/// relatively to the beginning of the path.
+#[derive(Serialize, Deserialize, ToSchema, PartialEq, Debug)]
+struct WorkScheduleProjection {
+    #[serde(rename = "type")]
+    #[schema(inline)]
+    pub work_schedule_type: WorkScheduleType,
+    pub start_date_time: NaiveDateTime,
+    pub end_date_time: NaiveDateTime,
+    pub path_position_ranges: Vec<(u64, u64)>,
+}
```

Comment:
Rather than documenting each field in the structure's documentation. The fields should be documented directly. 
You can run `cargo doc --open` to see how it looks.

```suggestion
/// A work schedule projected on a path
struct WorkScheduleProjection {
    /// The work schedule type
    #[serde(rename = "type")]
    #[schema(inline)]
    pub work_schedule_type: WorkScheduleType,
    /// The date and time when the work schedule takes effect.
    pub start_date_time: NaiveDateTime,
    /// The date and time at which the work schedule begins to apply
    pub end_date_time: NaiveDateTime,
    /// The date and time at which the work schedule ends.
    pub path_position_ranges: Vec<(u64, u64)>,
}
```

Reply:
done